airflow/models/taskinstance.py

```diff
@@ -172,6 +172,7 @@ def clear_task_instances(
                 # original max_tries or the last attempted try number.
                 ti.max_tries = max(ti.max_tries, ti.prev_attempted_tries)
             ti.state = State.NONE
+            ti.external_executor_id = None
```

Comment:
Could you check this in the unit tests so we don't regress

Reply:
Good one! Will add it.

Reply:
Updated the current test to also verify the `external_executor_id` is reset.